conda_forge_tick/contexts.py

```diff
@@ -0,0 +1,72 @@
+import copy
+from dataclasses import dataclass
+from networkx import DiGraph
+import typing
+import threading
+import github3
+
+if typing.TYPE_CHECKING:
+    from conda_forge_tick.migrators import Migrator
+    from conda_forge_tick.migrators_types import AttrsTypedDict
+
+
+@dataclass
+class GithubContext:
+    github_username: str
+    github_password: str
+    circle_build_url: str
+    github_token: typing.Optional[str] = ""
+    dry_run: bool = True
+    _tl: threading.local = threading.local()
+
+    @property
+    def gh(self) -> github3.GitHub:
+        if getattr(self._tl, "gh") is None:
+            if self.github_token:
+                gh = github3.login(token=self.github_token)
+            else:
+                gh = github3.login(self.github_username, self.github_password)
+            setattr(self._tl, "gh", gh)
+        return self._tl.gh
+
+
+@dataclass
+class MigratorsContext(GithubContext):
+    graph: DiGraph = None
+    smithy_version: str = ""
+    pinning_version: str = ""
+    prjson_dir = "pr_json"
+    rever_dir: str = "./feedstocks/"
+    quiet = True
+
+
+@dataclass
+class MigratorContext:
+    parent: MigratorsContext

```

Comment:
Maybe `migrators_ctx`? Since we will have full refactor tools we can use long names!

Reply:
Or maybe `migration_session_ctx`